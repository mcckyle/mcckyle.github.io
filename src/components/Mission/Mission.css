/* Filename: Mission.css */

/* =======================
   Mission Styles.
========================== */
#mission .section-title
{
    color: var(--secondary);
}
