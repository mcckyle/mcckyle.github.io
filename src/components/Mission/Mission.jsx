//Filename: Mission.jsx
//Author: Kyle McColgan
//Date: 30 October 2025
//Description: This file contains the component for the personal React project mission statement.

import React from 'react';
import './Mission.css';

function Mission()
{
    return (
      <div
        id="mission"
        role="region"
        aria-labelledby="mission-title"
        className="section-content"
      >
        <h2 id="mission-title" className="section-title">
          Mission Statement
        </h2>
        <div id="mission-body" className="section-body">
          <p>
            My mission is to uplift the local tech community by sharing the
            knowledge I've gained and fostering an environment of integrity,
            collaboration, and continuous growth. Through leading by example,
            I aim to inspire others in their journey toward technical excellence.
          </p>
          <p>
            I am dedicated to supporting developers, IT professionals, and
            organizations in building secure, efficient, and scalable systems. My
            goal is to exceed expectations by delivering solutions that are
            thoughtfully designed, secure, and accessible.
          </p>
          <p>
            Driven by the evolving world of technology, I embrace challenges as
            opportunities for growth. By staying curious and persistent, I strive
            to contribute positively to the broader tech landscape while growing
            personally and professionally.
          </p>
        </div>
      </div>
    );
}

export default Mission;
