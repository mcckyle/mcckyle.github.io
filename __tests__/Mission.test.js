//Filename: Mission.test.js
//Author: Kyle McColgan
//Date: 28 October 2025
//Description: This file contains the Jest unit tests for the personal React project mission section.

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Mission from '../src/components/Mission/Mission.jsx';

describe('Mission Component', () => {
    beforeEach(() => {
        render(<Mission />);
    });

    //Test #1
    it('renders the Mission section', () => {
        const section = screen.getByRole('region', { name: /Mission Statement/i });
        expect(section).toBeInTheDocument();
    });

    //Test #2
    it('renders the section title correctly', () => {
        const title = screen.getByRole('heading', { level: 2 });
        expect(title).toHaveTextContent('Mission Statement');
    });

    //Test #3
    it('renders all paragraphs in the Mission section', () => {
        const paragraphs = screen.getAllByText(/I/);
        expect(paragraphs).toHaveLength(3);
    });

    //Test #4
    it('renders the first paragraph correctly', () => {
        const firstParagraph = screen.getByText(/My mission is to uplift/);
        expect(firstParagraph).toBeInTheDocument();
    });

    //Test #5
    it('renders the second paragraph correctly', () => {
        const secondParagraph = screen.getByText(/I am dedicated to supporting/);
        expect(secondParagraph).toBeInTheDocument();
    });

    //Test #6
    it('renders the third paragraph correctly', () => {
        const thirdParagraph = screen.getByText(/Driven by the evolving world of technology/);
        expect(thirdParagraph).toBeInTheDocument();
    });

    //Test #7
    it('has the correct ARIA role of "region"', () => {
        const section = screen.getByRole('region', { name: /Mission Statement/i });
        expect(section).toBeInTheDocument();
    });

    //Test #8
    it('matches the snapshot of the Mission section', () => {
        const { asFragment } = render(<Mission />);
        expect(asFragment()).toMatchSnapshot();
    });

    //Test #9
    it('ensures there are no empty paragraphs', () => {
        const paragraphs = screen.getAllByText(/I/);
        paragraphs.forEach(paragraph => {
            expect(paragraph).not.toBeEmptyDOMElement();
        });
    });

    //Test #10
    it('does not log any errors to the console', () => {
        const consoleError = jest.spyOn(console, 'error').mockImplementation();
        expect(consoleError).not.toHaveBeenCalled();
    });
});
