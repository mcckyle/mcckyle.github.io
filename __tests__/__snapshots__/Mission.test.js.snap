// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Component matches the snapshot of the Mission section 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="mission-title"
    class="section"
    id="mission"
  >
    <h2
      class="section-title"
      id="mission-title"
    >
      Mission Statement
    </h2>
    <div
      class="section-body"
      id="mission-body"
    >
      <p>
        My mission is to uplift the local tech community by sharing the knowledge I've gained and fostering an environment of integrity, collaboration, and continuous growth. Through leading by example, I aim to inspire others in their journey toward technical excellence.
      </p>
      <p>
        I am dedicated to supporting developers, IT professionals, and organizations in building secure, efficient, and scalable systems. My goal is to exceed expectations by delivering solutions that are thoughtfully designed, secure, and accessible.
      </p>
      <p>
        Driven by the evolving world of technology, I embrace challenges as opportunities for growth. By staying curious and persistent, I strive to contribute positively to the broader tech landscape while growing personally and professionally.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
